crates/storage/codecs/derive/src/compact/mod.rs

```diff
@@ -196,7 +196,7 @@ fn should_use_alt_impl(ftype: &str, segment: &syn::PathSegment) -> bool {
 pub fn get_bit_size(ftype: &str) -> u8 {
     match ftype {
         "TransactionKind" | "TxKind" | "bool" | "Option" | "Signature" => 1,
-        "TxType" => 2,
+        "TxType" | "OpTxType" => 2,
```

Comment:
this could be a footgun.

perhaps a better approach long term would be a trait, maybe.

Reply:
I've looked into this, we can't solve this with a trait while we depend on `modular_bitifield` the `get_bit_size` output is translates into `B2`/`B1` etc inputs for bitfield macro:
```rust
#[bitfield]
pub struct OpReceiptFlags {
    pub tx_type_len: B2,
    pub success_len: B1,
    pub cumulative_gas_used_len: B4,
    pub deposit_nonce_len: B1,
    pub deposit_receipt_version_len: B1,
    pub __zstd: B1,
    #[skip]
    unused: B6,
}
```